File: leetcode/solutions/3120.py
# https://leetcode.com/problems/count-the-number-of-special-characters-i/

class Solution:
    def numberOfSpecialChars(self, word: str) -> int:
        bitmask_lower = 0
        bitmask_upper = 0

        for char in word:
            if ord(char) >= 97:
                bitmask_lower |= (1 << ord(char) - 97)
            else:
                bitmask_upper |= (1 << ord(char) - 65)
        
        return (bitmask_lower & bitmask_upper).bit_count()

File: leetcode/solutions/3120.rs
// https://leetcode.com/problems/count-the-number-of-special-characters-i/

impl Solution {
    pub fn number_of_special_chars(word: String) -> i32 {
        let mut bitmask_lower: u32 = 0;
        let mut bitmask_upper: u32 = 0;

        for chr in word.bytes() {
            if chr.is_ascii_lowercase() {
                bitmask_lower |= (1 << chr - b'a');
            } else {
                bitmask_upper |= (1 << chr - b'A');
            }
        }

        (bitmask_lower & bitmask_upper).count_ones() as i32
    }
}
